apps/_example05app/containers/FormTableExcelImport.js

```diff
@@ -0,0 +1,249 @@
+    background: red;
+`
+
+const OkText = styled.div`
+    word-break: break-word;
+    background: green;
+`
+
+const WarnText = styled.div`
+    word-break: break-word;
+    background: yellow;
+`
+
+function MappingForm ({ columns, cols, onChangeMapping, onFinish }) {
+    const [form] = Form.useForm()
+    const [values, setValues] = useState({})
+
+    const intl = useIntl()
+    const ColumnMsg = intl.formatMessage({ id: 'containers.FormTableExcelImport.Column' })
+    const SelectColumnMsg = intl.formatMessage({ id: 'containers.FormTableExcelImport.SelectColumn' })
+    const NextStepButtonLabelMsg = intl.formatMessage({ id: 'containers.FormTableExcelImport.NextStepButtonLabel' })
+
+    function handleChanges (changedValues, allValues) {
+        const titleToIndex = Object.fromEntries(cols.map((col) => [col.title, col.dataIndex]))
+        const remapped = Object.fromEntries(
+            Object.entries(allValues)
+                .filter(([k, v]) => v)
+                .map(([k, v]) => [titleToIndex[v], k]),
+        )
+        setValues(allValues)
+        onChangeMapping(remapped)
+    }
+
+    {/* disable chrome autofill hack 2 */}
+    useEffect(() => {
+        document.querySelectorAll('.ant-select-selector input').forEach((e) => {
+            e.setAttribute('autocomplete', `stopDamnAutocomplete${Math.random()}`)
+        })
+    })
+
+    const selectColumnComponent = (
+        <Select allowClear={true}
+                showSearch
+                placeholder={SelectColumnMsg}
+                optionFilterProp="children"
+                filterOption={(input, option) =>
+                    option.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
+                }
+        >
+            {cols.length > 0 ?
+                cols.map((col) => {
+                    if (Object.values(values).includes(col.title)) {
+                        return null
+                    }
+                    return <Select.Option key={col.title} value={col.title}>{col.title}</Select.Option>
+                })
+                :
+                null}
+            })
+        </Select>
+    )
+
+    return <Form
+        layout={'horizontal'}
+        size={'small'}
+        form={form}
+        onValuesChange={handleChanges}
+        onFinish={onFinish}
+    >
+        {/* disable chrome autofill hack 1*/}
+        <Form.Item label="Hidden" name="name1" style={{ display: 'none' }}>
+            <Input/>
+        </Form.Item>
+        {columns.map((column, index) => {
+            return <Form.Item key={index} label={[ColumnMsg, ' "', column.title, '"']} name={column.dataIndex}>
+                {selectColumnComponent}
+            </Form.Item>
+        })}
+        <Form.Item>
+            <Button type="primary" htmlType="submit">{NextStepButtonLabelMsg}</Button>
+        </Form.Item>
+    </Form>
+}
+
+function ExcelExporterButton ({ columns, setExportedData }) {
+    if (!columns) throw new Error('no columns prop')
```

Comment:
Why not simply use [PropTypes](https://reactjs.org/docs/typechecking-with-proptypes.html) ? 

They allow us to explicitly specify the props, that are used in component and throw warnings if something is not right

Reply:
It's better to use PropTypes. I don't want to add PropTypes for some components it's better to add it to all public components. I think in the future we can refactor it. But regardless of this at the moment, we should avoid unclear errors for developers.